camel-core/src/main/docs/validator.adoc

```diff
@@ -0,0 +1,190 @@
+[source,java]
+---------------------------------------------
+scheme:name
+---------------------------------------------
+where *scheme* is the type of data model like `java`, `xml` or `json`, and *name* is the individual
+data type name.
+
+
+
+[[Validator-SupportedValidators]]
+Supported Validators
+^^^^^^^^^^^^^^^^^^^^^
+
+|=======================================================================
+| Validator | Description
+
+| Predicate Validator | Validate with using link:expression.html[Expression] or link:predicate.html[Predicate]
+
+| Endpoint Transformer | Validate by forwarding to the link:endpoint.html[Endpoint] to be used with validation component such as link:validator-component.adoc[Validation Component] or link:bean-validation.html[Bean Validation Component].
+
+| Custom Transformer | Validate with using custom validator class. Validator must be a subclass of `org.apache.camel.spi.Validator`
+|=======================================================================
+
+
+
+[[Validator-CommonOptions]]
+Common Options
+^^^^^^^^^^^^^
+All validators have following common options to specify which data type is supported by the validator.
+`type` must be specified.
+|=======================================================================
+| Name | Description
+
+| type | <<Validator-DataTypeFormat,Data type>> to validate.
+|=======================================================================
+
+
+
+[[Validator-Predicate]]
+Predicate Validator Options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+|=======================================================================
+| Name | Description
+
+| expression | link:expression.html[Expression] or link:predicate.html[Predicate] to be used for validation
+|=======================================================================
+Here is an example to specify a validation predicate:
+
+Java DSL:
+[source,java]
+-------------------------------------------------------------------
+validator()
+    .type("csv:CSVOrder")
+    .withExpression(bodyAs(String.class).contains("{name:XOrder}"));
+-------------------------------------------------------------------
+
+XML DSL:
+[source,xml]
+-------------------------------------------------------------------
+<predicateValidator Type="csv:CSVOrder">
+    <simple>${body} contains '{name:XOrder'</simple>
+</dataFormatTransformer>
+-------------------------------------------------------------------
+
+
+
+[[Validator-Endpoint]]
+Endpoint Validator Options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+|=======================================================================
+| Name | Description
+
+| ref | Reference to the link:endpoint.html[Endpoint] ID
+
+| uri | link:endpoint.html[Endpoint] URI
+|=======================================================================
+Here is an example to specify endpoint URI in Java DSL:
+[source,java]
+-------------------------------------------------------------------
+validator()
+    .type("xml")
+    .withUri("validator:xsd/schema.xsd");
```

Comment:
In this example we may want to highlight that the `validator` endpoint being used is the Camel validator component that has been with Camel for a very long time. And that you can use any other Camel endpoint if you want to such as bean-valdator or any custom component